client/src/main/scala/org/apache/celeborn/client/WorkerStatusTracker.scala

```diff
@@ -40,6 +40,7 @@ class WorkerStatusTracker(
 
   val excludedWorkers = new ShuffleFailedWorkers()
   val shuttingWorkers: JSet[WorkerInfo] = new JHashSet[WorkerInfo]()
+  val availableWorkers: JSet[WorkerInfo] = new JHashSet[WorkerInfo]()
```

Comment:
availableWorkers should be `workersWithEndpoints` and we can keep unResolvedWorkers for newly workers.